mlir/lib/Quantum/Utils/QuantumSplitting.cpp

```diff
@@ -89,17 +91,73 @@ void AugmentedCircuitGenerator::generate(Region &region, OpBuilder &builder)
         else if (auto extractOp = dyn_cast<quantum::ExtractOp>(op)) {
             cacheDynamicWire(extractOp, builder);
         }
-        else if (auto gate = dyn_cast<quantum::DifferentiableGate>(op)) {
-            ValueRange diffParams = gate.getDiffParams();
-            if (!diffParams.empty()) {
-                for (Value param : diffParams) {
-                    builder.create<ListPushOp>(gate.getLoc(), oldToCloned.lookupOrDefault(param),
-                                               cache.paramVector);
+        else if (auto gate = dyn_cast<quantum::ParametrizedGate>(op)) {
+            ValueRange params = gate.getAllParams();
+
+            for (Value param : params) {
+                Location loc = gate.getLoc();
+                Value clonedParam = oldToCloned.lookupOrDefault(param);
+                Type paramType = clonedParam.getType();
+
+                if (!paramType.dyn_cast<RankedTensorType>()) {
+                    builder.create<ListPushOp>(loc, clonedParam, cache.paramVector);
+                    continue;
+                }
+
+                assert(isa<RankedTensorType>(paramType) && "Parameter type cannot be handled.");
+                Type aType = param.getType();
+                // Invariants, this is developer documentation.
+                // aType must be a tensor<NxNxf64>
+                auto aTensor = aType.cast<RankedTensorType>();
+                ArrayRef<int64_t> shape = aTensor.getShape();
+                assert(shape.size() == 2 && "Unexpected tensor shape in QubitUnitaryOp");
+                assert(shape[0] == shape[1] && "QubitUnitaryOp is not square matrix");
+                assert(shape[0] > 1 && "QubitUnitaryOp has invalid tensor shape.");
```

Comment:
Rather than use assertions (or on top of the assertions) I think we should raise pass failures or similar for any cases that aren't supported.
If right now we support:
- `f64`
- `tensor<NxNxcomplex<f64>>`

Than any other data type in the parameters should gracefully raise an error.